patches/server/0906-implement-real-multiplication-operation.patch

```diff
@@ -0,0 +1,43 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: DaRacci <racci@racci.dev>
+Date: Sat, 28 May 2022 23:44:17 +1000
+Subject: [PATCH] implement real multiplication operation
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/ai/attributes/AttributeInstance.java b/src/main/java/net/minecraft/world/entity/ai/attributes/AttributeInstance.java
+index 9c58d1190377ca887aa703cb8a96a2b028b63af1..c57578d3b8f66ed1bf355939c6a9340b9ec7ebcc 100644
+--- a/src/main/java/net/minecraft/world/entity/ai/attributes/AttributeInstance.java
++++ b/src/main/java/net/minecraft/world/entity/ai/attributes/AttributeInstance.java
+@@ -147,7 +147,11 @@ public class AttributeInstance {
+         for(AttributeModifier attributeModifier3 : this.getModifiersOrEmpty(AttributeModifier.Operation.MULTIPLY_TOTAL)) {
+             e *= 1.0D + attributeModifier3.getAmount();
+         }
+-
++        // Paper start - implement proper multiplication operation
++        for(AttributeModifier attributeModifier4 : this.getModifiersOrEmpty(AttributeModifier.Operation.MULTIPLY_TOTAL_REAL)) {
++            e *= attributeModifier4.getAmount();
++        }
++        // Paper end
+         return this.attribute.sanitizeValue(e);
+     }
+ 
+diff --git a/src/main/java/net/minecraft/world/entity/ai/attributes/AttributeModifier.java b/src/main/java/net/minecraft/world/entity/ai/attributes/AttributeModifier.java
+index 09b8fa2cee63e6bcb92ffc3694f67d0f90a3805d..f49a449bfa7f5372cf89680dfea6db637fce4b18 100644
+--- a/src/main/java/net/minecraft/world/entity/ai/attributes/AttributeModifier.java
++++ b/src/main/java/net/minecraft/world/entity/ai/attributes/AttributeModifier.java
+@@ -98,9 +98,13 @@ public class AttributeModifier {
+     public static enum Operation {
+         ADDITION(0),
+         MULTIPLY_BASE(1),
+-        MULTIPLY_TOTAL(2);
++        // Paper start - implement proper multiplication operation
++        MULTIPLY_TOTAL(100),
```

Comment:
Only the custom operation type should be shifted, not the vanilla one. Also the bukkit javadoc is missing for the new type

Reply:
That would appear to be a mess up on my end, ill fix that lol